### PR TITLE
Add implementation for dormant users engagement

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/19295

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Comment by kjozwiak** [comment]
Removing the `1.33.x - Beta` milestone has this hasn't even landed. Once landed, it will be in `1.34.x` and an uplift for `1.33.x` & `1.32.x` will needed to be created.

File: android/java/org/chromium/chrome/browser/DormantUsersEngagementDialogFragment.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) 2020 The Brave Authors. All rights reserved.
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/DormantUsersEngagementDialogFragment.java:2]: 2021

**Comment by SergeyZhukovsky** [comment]
Please check it on Android 6 device as I see you are using gradient there in vector images. We had problems with it in the past

File: android/java/org/chromium/chrome/browser/DormantUsersEngagementDialogFragment.java (hunk 1 above)
Comment (bsclifton) [inline android/java/org/chromium/chrome/browser/DormantUsersEngagementDialogFragment.java:2]:
```suggestion
 * Copyright (c) 2021 The Brave Authors. All rights reserved.
```

File: android/java/org/chromium/chrome/browser/app/BraveActivity.java [hunk 2]
```diff
@@ -560,6 +561,60 @@ public void finishNativeInitialization() {
                         BraveVpnUtils.SUBSCRIPTION_PARAM_TEXT, getPackageName());
             }
         }
+        if (OnboardingPrefManager.getInstance().isDormantUsersEngagementEnabled()
+                || getPackageName().equals(BRAVE_PRODUCTION_PACKAGE_NAME)) {
+            OnboardingPrefManager.getInstance().setDormantUsersNotificationTime(
+                    RetentionNotificationUtil.DORMANT_USERS_DAY_14, setTimeInMillis(14 * 24 * 60));
```
Comment (bsclifton) [inline android/java/org/chromium/chrome/browser/app/BraveActivity.java:567]: is this correct? 14 * 24 * 60 is correct # of seconds; but for milliseconds it needs to be `* 1000`
(same with below)

Comment (deeppandya) [reply android/java/org/chromium/chrome/browser/app/BraveActivity.java:567]: @bsclifton that's correct as we pass value to setTimeInmiilis in minutes and then in that function we need to convert it in millis for notifications.

Comment (bsclifton) [reply android/java/org/chromium/chrome/browser/app/BraveActivity.java:561]: ah ok - makes sense; `setTimeInMillis` is adding that many minutes 👍

**Review by bsclifton — APPROVED** [review]
++ from me

**Pull request merged**